src/Microsoft.Data.SqlClient/tests/ManualTests/ProviderAgnostic/MultipleResultsTest/MultipleResultsTest.cs

```diff
@@ -2,366 +2,154 @@
-            var twriter = new CarriageReturnLineFeedReplacer(swriter);
-            Console.SetOut(twriter); // "redirect" Console.Out
-
-            // Run Test
-            RunTest();
-
-            Console.Out.Flush();
-            Console.Out.Dispose();
-
-            // Recover the standard output stream
-            StreamWriter standardOutput = new StreamWriter(Console.OpenStandardOutput());
-            standardOutput.AutoFlush = true;
-            Console.SetOut(standardOutput);
-
-            // Compare output file
-            var comparisonResult = FindDiffFromBaseline(baselinePath, outputPath);
-
-            if (string.IsNullOrEmpty(comparisonResult))
-            {
-                return true;
-            }
-
-            Console.WriteLine("Test Failed!");
-            Console.WriteLine("Please compare baseline : {0} with output :{1}", Path.GetFullPath(baselinePath), Path.GetFullPath(outputPath));
-            Console.WriteLine("Comparison Results : ");
-            Console.WriteLine(comparisonResult);
-            return false;
-        }
+            command.CommandText = s_sqlStatement;
 
-        private void PrintException(Type expected, Exception e, params string[] values)
-        {
-            try
-            {
-                Debug.Assert(e != null, "PrintException: null exception");
-
-                _globalBuilder.Length = 0;
-                _globalBuilder.Append(e.GetType().Name).Append(": ");
-
-                if (e is COMException)
-                {
-                    _globalBuilder.Append("0x").Append((((COMException)e).HResult).ToString("X8"));
-                    if (expected != e.GetType())
-                    {
-                        _globalBuilder.Append(": ").Append(e.ToString());
-                    }
-                }
-                else
-                {
-                    _globalBuilder.Append(e.Message);
-                }
-                AssemblyFilter(_globalBuilder);
-                Console.WriteLine(_globalBuilder.ToString());
-
-                if (expected != e.GetType())
-                {
-                    Console.WriteLine(e.StackTrace);
-                }
-                if (values != null)
-                {
-                    foreach (string value in values)
-                    {
-                        Console.WriteLine(value);
-                    }
-                }
-                if (e.InnerException != null)
-                {
-                    PrintException(e.InnerException.GetType(), e.InnerException);
-                }
-                Console.Out.Flush();
-            }
-            catch (Exception f)
-            {
-                Console.WriteLine(f);
-            }
+            // ExecuteNonQuery will drain every result set, info message and exception, collating these into a single exception.
+            SqlException exNonQuery = Assert.Throws<SqlException>(() => command.ExecuteNonQuery());
```

Comment:
I'm probably going to sound super pedantic here, but it would be really nice to use this as an opportunity to clean up the tests a wee bit more. I try to write my tests in the Arrange/Act/Assert style, when possible. In this case, we're doing assert at the same time as acting, which makes it a bit difficult to see where the test action is taking place. If possible, can we extract the `command.ExecuteNonQuery()` to a variable (in nUnit, it'd be a `TestDelegate` type, but I think xUnit allows for `Action`)?

Reply:
I think I see what you mean, `Assert.Throws` blurs the line a little. I've separated this out.